src/duckxiterator.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Under MIT license
+ * Author: Amir Mohamadi (@amiremohamadi)
+ * DuckX is a free library to work wirh docx files.
+ */
+
+#ifndef DUCKXITERATOR_H
+#define DUCKXITERATOR_H
+
+namespace duckx {
+	template <class T>
+	class Iterator {
+	private:
+		T* _obj = nullptr;
+
+	public:
+		Iterator() = default;
+		Iterator(T& obj) : _obj(&obj) {}
+
+		// We only care comparator against end.
+		bool operator!=(const Iterator&) const { return _obj != nullptr; }
```

Comment:
Why not just compare against other._obj this way we can compare against subranges.

Reply:
I just used an iterator implementation I have seen elsewhere that would be a viable wrapper. I am resolving this conversation, because input iterator conversation already contains an iterator comparison requirement.